test-all-scream significant refactor

jgfouca opened the pull request:
Refactored items:
1) Tests become a proper object that can store test state
2) Valgrind and cuda-mem-check become test modifiers instead of test types
3) Use python f-strings instead of .format
4) 'opt' test no longer is set to do baseline checking. This made no sense
   since release mode is not BFB

jgfouca commented:
@AaronDonahue adding you since we discussed this briefly on slack. This PR fixes the issue of undocumented test types in the `--help`. I also decided that valgrind and cuda-memcheck should be modifiers instead of test types. I think this is better because it would allow the user the flexibility to do things like a valgrind single-precision test (with valgrind as a test type, this was not possible).

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 2861</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 3489</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Blake</h3>
<ul>
  <li>Build Num: 3608</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: SCREAM (E3SM-Project/scream)</dt>
  <ul>
    <li>Branch: jgfouca/tas_refactor</li>
    <li>SHA: 628a2f21a089cc88db20baa8e0c8c3a335b049d8</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: jgfouca
</details>

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 2861</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 3489</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Blake</h3>
<ul>
  <li>Build Num: 3608</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>628a2f21a089cc88db20baa8e0c8c3a335b049d8</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>
</details>

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - - This Pull Request Requires Inspection... The code must be inspected by a member of the Team before Testing/Merging
           NO REVIEWS HAVE BEEN PERFORMED ON THIS PULL REQUEST!

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

E3SM-Autotester commented:
All Jobs Finished; status = PASSED, However Inspection must be performed before merge can occur...

ambrad commented:
@jgfouca would you expand on this statement?
> 'opt' test no longer is set to do baseline checking. This made no sense since release mode is not BFB

My understanding is that baseline checking means that a run of a test is checked against reference output. The reference output is updated manually when a non-BFB PR is merged. According to this understanding, every build type should do baseline checking.

bartgol reviewed (changes requested):
Looks good. I have a suggestion about memcheck/valgrind.

As for baselines, we should be able to do _some_ baselines checking also in opt mode. The only potentially non deterministc ops are sum-reductions (max/min reductions should be ok). I see P3 has no such reduction, so even in OPT should generate the same result across runs, no?

ambrad commented:
@bartgol SCREAM as a whole, and thus all subcomponents, is run-to-run deterministic in all builds at this point, and we want to maintain that property.

jgfouca commented:
@ambrad , @bartgol , my comment regarding opt testing and baselines was based on this:

`src/share/scream_types.hpp`:
```
// We cannot expect BFB results between f90 and cxx if optimizations are on.                                                                                                                                                                   
// Same goes for cuda-memcheck because it makes the bfb math layer prohibitively                                                                                                                                                               
// expensive and so must be turned off.                                                                                                                                                                                                        
#if defined (NDEBUG) || defined (EKAT_ENABLE_CUDA_MEMCHECK)
static constexpr bool SCREAM_BFB_TESTING = false;
#else
static constexpr bool SCREAM_BFB_TESTING = true;
#endif
```

`src/physics/p3/tests/p3_run_and_cmp.cpp`:
```
scream::Real tol = SCREAM_BFB_TESTING ? 0 : std::numeric_limits<Real>::infinity();
```

`src/physics/shoc/tests/shoc_run_and_cmp.cpp`
```
scream::Real tol = SCREAM_BFB_TESTING ? 0 : std::numeric_limits<Real>::infinity();
```

The comment in scream_types.hpp for SCREAM_BFB_TESTING is incomplete. It is true that SCREAM_BFB_TESTING controls the f90/cxx bfb unit testing, but it also sets the default tolerance for the shoc/p3 run_and_cmp programs to infinity, which essentially disables the baseline checks. Maybe we shouldn't be doing that anymore?

In any event, I will turn baselines back on in opt mode so we will preserve the original (before the PR) behavior.

ambrad commented:
I see. To be clear, then, when you refer to baselines you mean only the standalone p3/shoc_run_and_cmp baseline checking, which also effectively is F90 vs C++ checking? Whatever you do is fine, then. I just wanted to check that this change doesn't affect CIME baseline testing in opt builds.

jgfouca commented:
@ambrad , correct, no impacts to anything CIME related. This is for internal testing and SCREAM_BFB_TESTING turns off the bfb unit test checking (comparing f90 and cxx) and the run_and_cmp_(f90|cxx) baseline checking which is NOT cross-language (run_and_cmp_cxx checks cxx runs against cxx baselines).

@bartgol , I looked over some old commits and the rationale for having baseline checking on for Release was to ensure that things (run_and_cmp, bfb unit tests) at least ran without exception/segfault even if exact results were not checked. By that logic, it seems like the FPE test should also do baseline checking too? It was set to off for some reason.

bartgol commented:
I think we should change the name then. `baseline` makes me think of run-to-run comparison of the _same_ implementation (be that f90 or cxx). The tests that have no hope to pass in opt are f90 vs cxx. Perhaps we should find a better name for those. I would vote for something very explicit, like `f90_vs_cxx`. Or maybe `cross_impl_testing` (more general, can be used for cxx-GPU vs cxx-CPU).

jgfouca commented:
@bartgol , I think I've addressed all your feedback for this PR. I did not do any further investigation into the deeper issue of whether run_and_cmp tolerances should be infinite in Release mode.

bartgol reviewed (approved):
We can address the issue of `run-to-run` vs `cross-impl` bfb-ness in a second PR.

E3SM-Autotester commented:
Status Flag 'Pre-Test Inspection' - Auto Inspected - Inspection Is Not Necessary for this Pull Request.

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Testing Jenkins Projects:

<details>
<summary>
  <b>Pull Request Auto Testing STARTING</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 2864</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 3492</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Blake</h3>
<ul>
  <li>Build Num: 3611</li>
  <li>Status: STARTED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Using Repos:</h2>

<dl>
  <dt>Repo: SCREAM (E3SM-Project/scream)</dt>
  <ul>
    <li>Branch: jgfouca/tas_refactor</li>
    <li>SHA: 0e668c572436a2e3f35730415afea895859a8f48</li>
    <li>Mode: TEST_REPO</li>
  </ul>
</dl>

Pull Request Author: jgfouca
</details>

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Jenkins Testing: all Jobs PASSED

<details>
<summary>
  <b>Pull Request Auto Testing has PASSED</b> (click to expand)
</summary>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Mappy</h3>
<ul>
  <li>Build Num: 2864</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Weaver</h3>
<ul>
  <li>Build Num: 3492</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>

<h2>Build Information</h2>
<h3>Test Name: SCREAM_PullRequest_Autotester_Blake</h3>
<ul>
  <li>Build Num: 3611</li>
  <li>Status: PASSED</li>
</ul>

<h4>Jenkins Parameters</h4>
<table>
  <tr>
    <th>Parameter Name</th>
    <th>Value</th>
  </tr>

  <tr>
    <td>PR_LABELS</td>
    <td>AT: AUTOMERGE;scripts</td>
  </tr>

  <tr>
    <td>PULLREQUESTNUM</td>
    <td>1814</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_BRANCH</td>
    <td>jgfouca/tas_refactor</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_SOURCE_SHA</td>
    <td>0e668c572436a2e3f35730415afea895859a8f48</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_BRANCH</td>
    <td>master</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_REPO</td>
    <td>https://github.com/E3SM-Project/scream</td>
  </tr>

  <tr>
    <td>SCREAM_TARGET_SHA</td>
    <td>55bebc98c9ed0bd37e2419fcfc65b4ba69443849</td>
  </tr>

  <tr>
    <td>TEST_REPO_ALIAS</td>
    <td>SCREAM</td>
  </tr>
</table>
</details>

E3SM-Autotester commented:
Status Flag 'Pre-Merge Inspection' - SUCCESS: The last commit to this Pull Request has been INSPECTED AND APPROVED by [ bartgol ]!

E3SM-Autotester commented:
Status Flag 'Pull Request AutoTester' - Pull Request will be Automerged

E3SM-Autotester commented:
Merge on Pull Request# 1814: IS A SUCCESS - Pull Request successfully merged
